interface/src/app/download/download.js

```diff
@@ -1,80 +1,74 @@
 /**
- * "download" module, which will download sample annotations of given
- * annotation typenames.
+ * "adage.download" module, which will download sample annotations based on
+ * selected annotation types.
  */
 
 angular.module('adage.download', [
   'ui.router',
-  'placeholders',
   'ui.bootstrap',
+  'ngResource',
   'as.sortable'
 ])
 
-.config(function config($stateProvider) {
+.config(['$stateProvider', function config($stateProvider) {
   $stateProvider.state('download', {
     url: '/download',
     views: {
-      'main': {
-        controller: 'DownloadCtrl',
-        templateUrl: 'download/download.tpl.html'
+      main: {
+        templateUrl: 'download/download.tpl.html',
+        controller: 'DownloadCtrl as ctrl'
       }
     },
     data: {pageTitle: 'Download'}
   });
-})
+}])
 
-// "AnnotationTypes" is a service that retrieves a list of available
-// annotation columns, from which the user can select for download.
 .factory('AnnotationTypes', ['$resource', function($resource) {
-  return $resource(
-    '/api/v0/annotationtype/',
-    {q: '', limit: 0},
-    {'query': {method: 'GET', isArray: false}}
-  );
+  return $resource('/api/v0/annotationtype');
 }])
 
-.controller('DownloadCtrl', ['$scope', '$window', 'AnnotationTypes',
-  function DownloadController($scope, $window, AnnotationTypes) {
-    $scope.annotations = {
-      query_message: 'Connecting to the server ...',
-      included_types: [],
-      excluded_types: []
-    };
-
-    AnnotationTypes.query(
-      {q: ''},
-      function(responseObject, responseHeaders) {
-        for (var i = 0; i < responseObject.objects.length; ++i) {
-          $scope.annotations.included_types.push(
-            responseObject.objects[i].typename);
-        }
-        $scope.annotations.query_message = '';
-      },
-      function(responseObject, responseHeaders) {
-        $log.error('Query errored with: ' + responseObject);
-        $scope.annotations.query_message = 'Connection to server failed';
-      }
-    );
+.controller('DownloadCtrl', ['$window', '$log', 'AnnotationTypes',
+  function DownloadController($window, $log, AnnotationTypes) {
+    var self = this;
```

Comment:
Part of my concern with this is that self is actually *not* a part of JavaScript. It can be found in the browser context because the global context is the `window` object, and there is a `window.self`, which points back to `window`. Creating a `var self =...` will mask that global. I think that masking is fine, actually, but it just doesn't feel like a best practice solution to me.

It works, and I know people do it that way (including the authors of that book you cite) but I think what bothers me about it is that it doesn't really clarify what is going on.

Anyway, this is really a style issue and not something we should resolve in a PR. Moving on...